include/morphio/morphology.h

```diff
@@ -99,17 +100,17 @@ class Morphology
 
        If id == -1, the iteration will start at each root section, successively
     **/
-    depth_iterator depth_begin() const;
-    depth_iterator depth_end() const;
+    depth_iterator_t<Section, Morphology> depth_begin() const;
```

Comment:
I suppose the old denomination `depth_iterator it = morphology.depth_begin()` still works ?

Reply:
I don't follow

Reply:
My question is: is this change breaking the API or do we need to change morphokit as well ?

Reply:
Ah, sorry, I see.  Could you try it?  Otherwise, I can make an type alias that should allow it to compile.  The ABI will be different, though, so we can't just link it.

Reply:
morphokit does not compile because of this.

```
/home/bcoste/workspace/morphology/morphokit/src/iterators.cpp:14:115: error: no matching function for call to ‘morphio::upstream_iterator_t<morphio::Section>::upstream_iterator_t(const morphio::Morphology&)’
                                  const Morphology* morphology) : BaseIterator(base_morpho), _morphology(morphology)
                                                                                                                   ^
In file included from /home/bcoste/.local/include/morphio/section.h:5:0,
                 from /home/bcoste/workspace/morphology/morphokit/src/../include/morphokit/iterators.h:2,
                 from /home/bcoste/workspace/morphology/morphokit/src/iterators.cpp:1:
/home/bcoste/.local/include/morphio/section_iterators.hpp:190:5: note: candidate: morphio::upstream_iterator_t<SectionT>::upstream_iterator_t(const morphio::upstream_iterator_t<SectionT>&) [with SectionT = morphio::Section]
     upstream_iterator_t(const upstream_iterator_t& other)
```

 The fix looks simple but that means the API is broken. However, I don't think we have other C++ users so we might not care.
Or would the `using depth_iterator = depth_iterator_t<....>` be possible ?

Reply:
Hmm, did you try w/ the 3c424d7 version? I thought I added all the type aliases.

Also, it looks like it's the `upstream_iterator` that's failing, and I don't think it makes sense to upstream iterate a morphology, since there is no where to iterate to.  Is that call necessary?

Reply:
Yes, I tried on 3c424d7.

Reply:
Ok, just making sure.

What about the point that upstream_iterator(Morphology) doesn't really make sense?

Reply:
That absolutely makes no sense, yes

Reply:
Deleting the upstream_iterator(Morphology) fixes the compilation.